simulation/simulation_interface/src/conversions.cpp

```diff
@@ -596,91 +605,155 @@ auto toProto(
-      case autoware_auto_perception_msgs::msg::TrafficLight::AMBER:
-      case autoware_auto_perception_msgs::msg::TrafficLight::GREEN:
-        return true;
-      default:
-        return false;
-    }
-  };
-
   auto convert_color = [](auto color) {
     switch (color) {
       case autoware_auto_perception_msgs::msg::TrafficLight::RED:
-        return simulation_api_schema::TrafficLightState::LampState::RED;
+        return simulation_api_schema::TrafficLight_Color_RED;
       case autoware_auto_perception_msgs::msg::TrafficLight::AMBER:
-        return simulation_api_schema::TrafficLightState::LampState::YELLOW;
+        return simulation_api_schema::TrafficLight_Color_AMBER;
       case autoware_auto_perception_msgs::msg::TrafficLight::GREEN:
-        return simulation_api_schema::TrafficLightState::LampState::GREEN;
+        return simulation_api_schema::TrafficLight_Color_GREEN;
+      case autoware_auto_perception_msgs::msg::TrafficLight::WHITE:
+        return simulation_api_schema::TrafficLight_Color_WHITE;
       default:
-        return simulation_api_schema::TrafficLightState::LampState::UNKNOWN;
-    }
-  };
-
-  auto has_convertible_shape = [](auto && traffic_light) {
-    switch (traffic_light.shape) {
-      case autoware_auto_perception_msgs::msg::TrafficLight::LEFT_ARROW:
-      case autoware_auto_perception_msgs::msg::TrafficLight::RIGHT_ARROW:
-      case autoware_auto_perception_msgs::msg::TrafficLight::UP_ARROW:
-      case autoware_auto_perception_msgs::msg::TrafficLight::DOWN_ARROW:
-        return true;
-      default:
-        return false;
+        return simulation_api_schema::TrafficLight_Color_UNKNOWN_COLOR;
     }
   };
 
   auto convert_shape = [](auto shape) {
     switch (shape) {
+      case autoware_auto_perception_msgs::msg::TrafficLight::CIRCLE:
+        return simulation_api_schema::TrafficLight_Shape_CIRCLE;
       case autoware_auto_perception_msgs::msg::TrafficLight::LEFT_ARROW:
-        return simulation_api_schema::TrafficLightState::LampState::LEFT;
+        return simulation_api_schema::TrafficLight_Shape_LEFT_ARROW;
       case autoware_auto_perception_msgs::msg::TrafficLight::RIGHT_ARROW:
-        return simulation_api_schema::TrafficLightState::LampState::RIGHT;
+        return simulation_api_schema::TrafficLight_Shape_RIGHT_ARROW;
       case autoware_auto_perception_msgs::msg::TrafficLight::UP_ARROW:
-        return simulation_api_schema::TrafficLightState::LampState::UP;
+        return simulation_api_schema::TrafficLight_Shape_UP_ARROW;
+        /// @note Enums below are not supported yet in some platforms. I temporarily disabled them
+        //  case autoware_auto_perception_msgs::msg::TrafficLight::UP_LEFT_ARROW:
+        //    return simulation_api_schema::TrafficLight_Shape_UP_LEFT_ARROW;
+        //  case autoware_auto_perception_msgs::msg::TrafficLight::UP_RIGHT_ARROW:
+        //    return simulation_api_schema::TrafficLight_Shape_UP_RIGHT_ARROW;
```

Comment:
The current traffic_simulator signal-related code was written by me.
At the time, we were in the process of integrating AutowareAuto and AutowareIV as the current AutowareUniverse.

Since the traffic light-related message types were to be unified with those of AutowareAuto, I implemented the traffic signals in traffic_simulator to be compatible with the traffic light messages that were included in the autoware_auto_msgs perception directory.

The puzzling lack of arrow signals bothered me at the time, but upon investigation, I discovered that AutowareAuto's traffic light messages were apparently designed according to the [Vienna Convention on Road Signs and Signals](https://en.wikipedia.org/wiki/Vienna_Convention_on_Road_Signs_and_Signals). Therefore, the absence of UP_LEFT and UP_RIGHT arrows is an intended design.

Reply:
_(The above comments are not a request to you to change anything.)_

Reply:
I've just noticed that UP_LEFT and UP_RIGHT arrows have been added to the `autoware_auto_perception_msgs` in https://github.com/tier4/autoware_auto_msgs/pull/42

Reply:
_(There is no need to support UP_LEFT and UP_RIGHT in this pull request, but we will probably have to in the near future......)_